python/python/lance/ray/sink.py

```diff
@@ -15,10 +27,93 @@
 if TYPE_CHECKING:
     import pandas as pd
 
-__all__ = ["LanceDatasink"]
+__all__ = ["LanceDatasink", "LanceFragmentWriter", "LanceCommitter"]
+
+
+def _pd_to_arrow(
+    df: Union[pa.Table, "pd.DataFrame", Dict], schema: Optional[pa.Schema]
+) -> pa.Table:
+    """Convert a pandas DataFrame to pyarrow Table."""
+    from ..dependencies import _PANDAS_AVAILABLE
+    from ..dependencies import pandas as pd
+
+    if isinstance(df, dict):
+        return pa.Table.from_pydict(df, schema=schema)
+    if _PANDAS_AVAILABLE and isinstance(df, pd.DataFrame):
+        return pa.Table.from_pandas(df, schema=schema)
```

Comment:
do we want to do drop the metadata here? `pa.Table.from_pandas` will add metadata about the original df's schema

Reply:
do you mean passing `preserve_index=False`? or explicitly providing `schema=schema`?

Reply:
Done